src/lib/re.js

```diff
@@ -30,137 +32,129 @@ var $builtinmodule = function (name) {
+        var pat, str, captured, jsflags, regex;
+        var result, match, index, splits;
+
         Sk.builtin.pyCheckArgs("split", arguments, 2, 4);
         if (!Sk.builtin.checkString(pattern)) {
             throw new Sk.builtin.TypeError("pattern must be a string");
         }
-        ;
         if (!Sk.builtin.checkString(string)) {
             throw new Sk.builtin.TypeError("string must be a string");
         }
-        ;
         if (maxsplit === undefined) {
             maxsplit = 0;
         }
-        ;
         if (!Sk.builtin.checkNumber(maxsplit)) {
             throw new Sk.builtin.TypeError("maxsplit must be a number");
         }
-        ;
         if (flags === undefined) {
             flags = 0;
         }
-        ;
         if (!Sk.builtin.checkNumber(flags)) {
             throw new Sk.builtin.TypeError("flags must be a number");
         }
-        ;
 
         maxsplit = Sk.builtin.asnum$(maxsplit);
-        var pat = Sk.ffi.unwrapo(pattern);
-        var str = Sk.ffi.unwrapo(string);
+        pat = Sk.ffi.unwrapo(pattern);
+        str = Sk.ffi.unwrapo(string);
 
         // Convert pat from Python to Javascript regex syntax
         pat = convert(pat);
         //print("Pat: " + pat);
         //print("Str: " + str);
 
-        var captured = !(pat.match(/^\(.*\)$/) === null);
+        captured = !(pat.match(/^\(.*\)$/) === null);
         //print("Captured: ", captured);
 
-        var jsflags = getFlags(flags);
+        jsflags = getFlags(flags);
         //print("Flags: ", jsflags);
 
-        var regex = new RegExp(pat, jsflags);
+        regex = new RegExp(pat, jsflags);
 
-        var result = [];
-        var match;
-        var index = 0;
-        var splits = 0;
+        result = [];
+        match;
+        index = 0;
+        splits = 0;
         while ((match = regex.exec(str)) != null) {
-            //print("Matched '" + match[0] + "' at position " + match.index + 
+            //print("Matched '" + match[0] + "' at position " + match.index +
             //      "; next search at " + regex.lastIndex);
             if (match.index === regex.lastIndex) {
                 // empty match
                 break;
             }
-            ;
             result.push(new Sk.builtin.str(str.substring(index, match.index)));
             if (captured) {
                 // Add matching pattern, too
                 result.push(new Sk.builtin.str(match[0]));
             }
-            ;
             index = regex.lastIndex;
             splits += 1;
             if (maxsplit && (splits >= maxsplit)) {
                 break;
             }
-            ;
         }
-        ;
         result.push(new Sk.builtin.str(str.substring(index)));
 
         return new Sk.builtin.list(result);
-    });
+    };
+
+    _split.co_varnames = ["maxsplit", "flags"];
+    _split.co_numargs = 4;
+    _split.$defaults = [ new Sk.builtin.int_(0), new Sk.builtin.int_(0) ];
+
+    mod.split = Sk.nativejs.func(_split);
```

Comment:
This was the original way of allowing keyword arguments.  Look at ```native.js```.  It was done differently later, probably because people were unaware that it was there.  So instead of using/improving it, other strategies were employed.  Remember the ```re``` module was written a long time ago and hasn't needed to be touched...